Update Readme for simplified Kops 1.10 instructions

rifelpet opened the pull request:
[Kops 1.10](https://github.com/kubernetes/kops/releases/tag/1.10.0) was released 2 days ago which [adds support for managing the AWS IAM Authenticator](https://github.com/kubernetes/kops/pull/5197) via the cluster spec. This greatly simplifies the install instructions.

It does require Kops 1.10 and Kubernetes 1.10, so if theres a desire to maintain both sets of instructions I can update this PR to do so.

christopherhein commented on README.md:
```diff
@@ -122,46 +122,20 @@ This is useful if you want to authenticate as an IAM user directly or if you wan
-4. Update the Authenticator DaemonSet's state and output volumes to both use `/srv/kubernetes/aws-iam-authenticator/` for their `hostPath`s.
-5. Apply the DaemonSet and ConfigMap resource manifests to launch the Authenticator server on the cluster.
 
 *Note:* Certain Kops commands will overwrite the `ExecCredential` in kubeconfig so it may need to be restored manually. See [kubernetes/kops#5051](https://github.com/kubernetes/kops/issues/5051) for more information.
```
We should probably change this to say something like "If you are using Kops <1.10 …" instead of "Certain Kops" since the 1.10 Kops fixes this bug by supporting `client-go` 7.0

rifelpet replied:
It's kind of strange to provide instructions that require >=1.10 and then give a warning about behavior when using <1.10. Maybe we just remove the warning altogether if we're not going to provide instructions for <1.10 ?

christopherhein replied:
That's a fair point. Removing is probably the better option.

rifelpet commented:
Updated to remove outdated note

dhemeier commented:
Hey @rifelpet ,

I had currently tested to change to aws-authenticator on one of my clusters. You don't need to do a `rolling-update` on existing clusters anymore. The `aws-iam-authenticator` deployed directly after the `kops update cluster --yes`. So you could also remove the rolling-update task from the description.

dhemeier commented:
Sorry this is not correct, 

all deployments and folders are created on the master, but the certificates are only inserted on instance launch. So you need the `rolling-update`. 

But then Kops have a problem, they don't display a "needs update" message, so you can only deploy the master again with `kops rolling-update cluster --instance-group master-XX-XX-XX --force --yes`

Sorry for my mistake.

rifelpet commented:
@dhemeier Thanks for the heads up, I've updated the `rolling-update` command to include `--force`, and added `--yes` to both `update` and `rolling-update`.

somcsel commented on README.md:
```diff
@@ -122,49 +122,20 @@ This is useful if you want to authenticate as an IAM user directly or if you wan
-*Note:* Certain Kops commands will overwrite the `ExecCredential` in kubeconfig so it may need to be restored manually. See [kubernetes/kops#5051](https://github.com/kubernetes/kops/issues/5051) for more information.
 
+2. Apply the changes with `kops update cluster ${CLUSTER_NAME} --yes`.
+   If the cluster already exists, roll the cluster with `kops rolling-update cluster ${CLUSTER_NAME} --force --yes` in order to recreate the master nodes.
```
If only the master nodes need to be rolled, then use `kops rolling-update cluster ${CLUSTER_NAME} --instance-group ${MASTER_INSTANCE_GROUP_NAME} --force --yes` to prevent unnecessary recycling of the nodes.

Globegitter replied:
More generically one could also just do `kops rolling-update cluster ${CLUSTER_NAME} --instance-group-roles=Master --force --yes`

somcsel commented:
According to the [kops docs](https://github.com/kubernetes/kops/blob/master/docs/authentication.md#aws-iam-authenticator) a ConfigMap also needs to be created.  Without this, the cluster will not validate.

somcsel commented:
Perhaps it would be better to maintain one set of authoritative instructions in the kops git and just link to them from this README?

rifelpet commented:
@somcsel Agreed. I've updated this PR to just point to the kops documentation. I added some additional instructions over there so hopefully it should be sufficient on its own.

nckturner commented:
@christopherhein is this ok to merge?

christopherhein commented:
/lgtm

@nckturner I'm good with these updates.

rifelpet commented:
Is there any way we can get this merged? CC @nckturner

nckturner commented:
/approve
